src/libdrakvuf/win-processes.c

```diff
@@ -313,23 +313,37 @@ static unicode_string_t* win_get_process_full_name(drakvuf_t drakvuf, addr_t epr
 
 char* win_get_process_name(drakvuf_t drakvuf, addr_t eprocess_base, bool fullpath)
 {
-    if ( fullpath )
+    unicode_string_t* fullname = win_get_process_full_name( drakvuf, eprocess_base );
+
+    if (fullname && fullname->contents && strlen((const char*)fullname->contents) > 0)
     {
-        unicode_string_t* fullname = win_get_process_full_name( drakvuf, eprocess_base );
+        // Replace 'proc_data->name' with 'fullname->contents'
+        // Moving ownership of fullname->contents to name for later cleanup
+        char* name = (char*)fullname->contents;
 
-        if (fullname && fullname->contents && strlen((const char*)fullname->contents) > 0)
+        // ImageFileName size differs between kernel builds & versions,
+        // relying on it would sometimes yield incomplete process name.
+        if ( !fullpath )
         {
-            // Replace 'proc_data->name' with 'fullname->contents'
-            // Moving ownership of fullname->contents to name for later cleanup
-            char* name = (char*)fullname->contents;
-            g_free( (gpointer)fullname );
-            return name;
+            char** tokens = g_strsplit((char*)fullname->contents, "\\", -1);
+            int index = 0;
+
+            while (tokens && tokens[index+1] != NULL)
+                index++;
+
+            if (tokens)
+                name = g_strdup(tokens[index]);
+
+            g_strfreev(tokens);
         }
 
-        if (fullname)
-            vmi_free_unicode_str(fullname);
+        g_free( (gpointer)fullname );
+        return name;
     }
 
+    if (fullname)
+        vmi_free_unicode_str(fullname);
```

Comment:
@tklengyel  Although I'm confused as it is the same code flow; it uses g_free before returning the variable `name` and in case the first condition `if (fullname && fullname->contents && strlen((const char*)fullname->contents) > 0)` is not satisfied, it will continue to use `vmi_free_unicode_str(fullname)`